src/GSF.CarbonIntensity/src/Handlers/EmissionsHandler.cs

```diff
@@ -0,0 +1,39 @@
+using CarbonAware.Aggregators.CarbonAware;
+using CarbonAware.Exceptions;
+using CarbonAware.Tools.WattTimeClient;
+using CarbonAware.Tools.WattTimeClient.Configuration;
+using GSF.CarbonIntensity.Exceptions;
+using Microsoft.Extensions.Logging;
+
+namespace GSF.CarbonIntensity.Handlers;
+
+internal sealed class EmissionsHandler : IEmissionsHandler
+{
+    private readonly ILogger<EmissionsHandler> _logger;
+    private readonly ICarbonAwareAggregator _aggregator;
+
+    public EmissionsHandler(ILogger<EmissionsHandler> logger, ICarbonAwareAggregator aggregator)
+    {
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        _aggregator = aggregator ?? throw new ArgumentNullException(nameof(aggregator));
+    }
+
+    /// <inheritdoc />
+    public async Task<double> GetAverageCarbonIntensityAsync(string location, DateTimeOffset start, DateTimeOffset end)
+    {
+        var parameters = new CarbonAwareParametersBaseDTO {
+            Start = start,
+            End = end,
+            SingleLocation = location
+        };
+        try {
+            var result = await _aggregator.CalculateAverageCarbonIntensityAsync(parameters);
+            _logger.LogDebug("calculated average carbon intensity: {carbonIntensity}", result);
+            return result;
+        }
+        catch (Exception ex) when (ex is WattTimeClientException || ex is WattTimeClientHttpException || ex is LocationConversionException || ex is ConfigurationException)
```

Comment:
We considered it but decided to leave it to another pass of work since we may be revisiting exceptions more broadly within the sdk and therefore the work would delve into areas outside of the SDK. We implemented this solution as a basic working example for now.

Reply:
We just finally got together on this one.  CarbonAware for the package name seems to be the prefered discussion point.  The vision will be to build a toolset to enable carbon aware applications, which could also include schedulers, tools for graceful degredation in carbon aware situations etc.  Carbon Intensity is sort of the first step because getting this data helps to drive the desired behaviours.  Emissions is not the one to use, as this generally relates to carbon reporting platforms and emissions data platforms.

Reply:
I personally am for it to go under `CarbonAware` - seems like the best fit, considering Vaughan's input. Definitely not `Emissions`.

Reply:
Great, we'll use `CarbonAware` then